packages/xstate-react/src/createActorContext.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { createContext, useContext } from 'react';
+import { useInterpret } from './useInterpret';
+import { useActor } from './useActor';
+import { useSelector } from './useSelector';
+import { ActorRefFrom, AnyStateMachine, EmittedFrom, EventFrom } from 'xstate';
+
+export function createActorContext<TMachine extends AnyStateMachine>(
+  machine: TMachine
+): React.Context<ActorRefFrom<TMachine>> & {
+  // (props: { children: React.ReactNode }): React.ReactElement<any, any>;
+  useActor: () => [
+    EmittedFrom<ActorRefFrom<TMachine>>,
+    (event: EventFrom<TMachine>) => void
+  ];
+  useSelector: <T>(
+    selector: (snapshot: EmittedFrom<TMachine>) => T,
+    compare?: (a: T, b: T) => boolean
+  ) => T;
+  useContext: () => ActorRefFrom<TMachine>;
+  Provider: (props: {
+    children: React.ReactNode;
+  }) => React.ReactElement<any, any>;
+} {
+  const ReactContext = createContext<ActorRefFrom<TMachine> | null>(
+    null
+  ) as any; // TODO: fix types
+
+  const OriginalProvider = ReactContext.Provider;
+
+  function Provider({ children }: { children: React.ReactNode }) {
+    const actor = useInterpret(machine) as ActorRefFrom<TMachine>;
+
+    return <OriginalProvider value={actor}>{children}</OriginalProvider>;
+  }
+
+  Provider.displayName = `Provider(${machine.id})`;
+
+  ReactContext.useContext = () => {
```

Comment:
The name of this one is super generic - I wonder if we couldn't rename it to something like `useActorRef`? That would make us have both `useActorRef` and `useActor`.

I'm not sure if that particular proposal is better - but it feels like a better name for this one would be appreciated. Let's not bother with it too much if we can't figure out a better one though

Reply:
We can also omit this in favor of the very idiomatic:

```js
const actorRef = useContext(SomeContext);

// versus:
// const actorRef = SomeContext.useContext();
```

Reply:
We could, but then we'd have to also return that `SomeContext` to be usable with `React.useContext` (so we'd have to name it, even if that would be `.context`). 

I think I would prefer to keep this extra helper - especially since it "validates" if the thing was actually provided.

Reply:
> We could, but then we'd have to also return that `SomeContext` to be usable with `React.useContext` (so we'd have to name it, even if that would be `.context`).
> 
> I think I would prefer to keep this extra helper - especially since it "validates" if the thing was actually provided.

Yep, I was just about to implement this and realized we don't get any helpful errors if the Provider is omitted.